config/v1/types_dns.go

```diff
@@ -96,3 +103,44 @@ type DNSList struct {
 
 	Items []DNS `json:"items"`
 }
+
+// DNSPlatformType is a cloud infrastructure provider that supports
+// platform specific DNS configuration.
+// +kubebuilder:validation:Enum=AWS
+type DNSPlatformType PlatformType
+
+const (
+	// AWSDNSPlatformType represents Amazon Web Services.
+	AWSDNSPlatformType DNSPlatformType = "AWS"
+)
+
+// DNSPlatformSpec holds cloud-provider-specific configuration
+// for DNS administration.
+// +union
+// +kubebuilder:validation:XValidation:rule="has(self.platform) && self.platform == 'AWS' ?  has(self.aws) : !has(self.aws)",message="aws configuration is required when platform is AWS, and forbidden otherwise"
+type DNSPlatformSpec struct {
+	// type is the underlying infrastructure provider for the cluster.
+	// Allowed values: "AWS".
+	// DNSPlatformType is a subset of Infrastructure PlatformType.
+	//
+	// Individual components may not support all platforms,
+	// and must handle unrecognized platforms with best-effort defaults.
+	//
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	Type DNSPlatformType `json:"type"`
+
+	// aws contains DNS configuration specific to the Amazon Web Services cloud provider.
+	// +optional
+	AWS *AWSDNSSpec `json:"aws"`
```

Comment:
Should this be kept as a pointer (because it is in the discriminated union)?

Reply:
> Should this be kept as a pointer (because it is in the discriminated union)?

Yes.